position/utils_test.gno

```diff
@@ -0,0 +1,344 @@
+				})
+			} else {
+				uassert.NotPanics(t, func() {
+					checkDeadline(tc.deadline)
+				})
+			}
+		})
+	}
+}
+
+func TestAssertOnlyUserOrStaker(t *testing.T) {
+	tests := []struct {
+		name         string
+		originCaller std.Address
+		expected     bool
+	}{
+		{
+			name:         "Failure - Not User or Staker",
+			originCaller: consts.ROUTER_ADDR,
+			expected:     false,
+		},
+		{
+			name:         "Success - User Call",
+			originCaller: consts.ADMIN,
+			expected:     true,
+		},
+		{
+			name:         "Success - Staker Call",
+			originCaller: consts.STAKER_ADDR,
+			expected:     true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			std.TestSetOrigCaller(tc.originCaller)
+			assertOnlyUserOrStaker(std.PrevRealm())
+		})
+	}
+}
+
+func TestAssertOnlyNotHalted(t *testing.T) {
+	tests := []struct {
+		name     string
+		expected bool
+		panicMsg string
+	}{
+		{
+			name:     "Failure - Halted",
+			expected: false,
+			panicMsg: "[GNOSWAP-COMMON-002] halted || gnoswap halted",
+		},
+		{
+			name:     "Success - Not Halted",
+			expected: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.expected {
+				uassert.NotPanics(t, func() {
+					assertOnlyNotHalted()
+				})
+			} else {
+				std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+				common.SetHaltByAdmin(true)
+				uassert.PanicsWithMessage(t, tc.panicMsg, func() {
+					assertOnlyNotHalted()
+				})
+				common.SetHaltByAdmin(false)
+			}
+		})
+	}
+}
+
+func TestAssertOnlyValidAddress(t *testing.T) {
+	tests := []struct {
+		name     string
+		addr     std.Address
+		expected bool
+		errorMsg string
+	}{
+		{
+			name:     "Success - valid address",
+			addr:     consts.ADMIN,
+			expected: true,
+		},
+		{
+			name:     "Failure - invalid address",
+			addr:     "g1lmvrrrr4er2us84h2732sru76c9zl2nvknha8",
+			expected: false,
+			errorMsg: "[GNOSWAP-POSITION-011] invalid address || (g1lmvrrrr4er2us84h2732sru76c9zl2nvknha8)",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.expected {
+				uassert.NotPanics(t, func() {
+					assertOnlyValidAddress(tc.addr)
+				})
+			} else {
+				uassert.PanicsWithMessage(t, tc.errorMsg, func() {
+					assertOnlyValidAddress(tc.addr)
+				})
+			}
+		})
+	}
+}
+
+func TestAssertOnlyValidAddressWith(t *testing.T) {
+	tests := []struct {
+		name     string
+		addr     std.Address
+		other    std.Address
+		expected bool
+		errorMsg string
+	}{
+		{
+			name:     "Success - validation address check to compare with other address",
+			addr:     consts.ADMIN,
+			other:    std.Address("g17290cwvmrapvp869xfnhhawa8sm9edpufzat7d"),
+			expected: true,
+		},
+		{
+			name:     "Failure - two address is different",
+			addr:     "g1lmvrrrr4er2us84h2732sru76c9zl2nvknha8",
+			other:    "invalidAddress",
+			expected: false,
+			errorMsg: "[GNOSWAP-POSITION-011] invalid address || (g1lmvrrrr4er2us84h2732sru76c9zl2nvknha8)",
```

Comment:
This case doesn't panic because of two address are different, but second address is just invalid. I think second address should be different valid address.

Reply:
69236fcf2522fd76bed8575b2c45bd5bfad288c5